server/src/main/java/org/elasticsearch/plugins/internal/DocumentParsingSupplierPlugin.java

```diff
@@ -8,15 +8,13 @@
 
 package org.elasticsearch.plugins.internal;
 
-import java.util.function.Supplier;
-
 /**
- * An internal plugin that will return a supplier of DocumentParsingObserver.
+ * An internal plugin that will return a supplier of DocumentParsingSupplier.
  */
-public interface DocumentParsingObserverPlugin {
+public interface DocumentParsingSupplierPlugin {
```

Comment:
Since we now call the supplier a provider, can we use the same terminology in the plugin here, ie DocumentParsingProviderPlugin?